Source/WebCore/rendering/RenderLayer.h

```diff
@@ -1233,6 +1237,11 @@ class RenderLayer final : public CanMakeSingleThreadWeakPtr<RenderLayer>, public
     bool has3DTransformedDescendant() const { ASSERT(!m_3DTransformedDescendantStatusDirty); return m_has3DTransformedDescendant; }
     bool has3DTransformedAncestor() const { return m_has3DTransformedAncestor; }
 
+    bool isViewportConstrained() const { return renderer().isFixedPositioned() || renderer().isStickilyPositioned(); }
```

Comment:
Thanks for the reviews! Going to make a tiny adjustment here, and make `isViewportConstrained` public on `RenderLayer`, so that I can use it in my next performance improvement PR.